torch/csrc/jit/script/compiler.cpp

```diff
@@ -2057,19 +2059,37 @@ struct to_ir {
       }
       return std::make_shared<SimpleValue>(expr);
     } else if(auto getattr = dynamic_cast<GetAttrValue*>(sv.get())) {
-      if (apply.attributes().size() > 0) {
-        throw ErrorReport(loc) << "getattr takes no keyword arguments";
-      }
-      if (apply.inputs().size() != 2) {
-        throw ErrorReport(loc) << "getattr expects 2 inputs";
-      }
+      checkApplyExpr(apply, loc);
       auto obj = emitSugaredExpr(apply.inputs()[0], 1);
       auto selector = apply.inputs()[1];
       if (selector.kind() != TK_STRINGLITERAL) {
         throw ErrorReport(loc) << "getattr's second argument must be a string literal";
       }
       const std::string& name = StringLiteral(selector).text();
       return obj->attr(apply.range(), method, name);
+    } else if (auto isinst = dynamic_cast<IsInstValue*>(sv.get())) {
+      // NOTE: for `isinstance` builtin call in JIT, we only check the static types
+      // on the inputs to evaluate, and insert the corresponding constant node
+      checkApplyExpr(apply, loc);
+      auto type_name = parseBaseTypeName(apply.inputs()[1]);
+      if (!type_name) {
+        throw ErrorReport(apply.inputs()[1].range()) << "type must be a type identifier";
+      }
+      auto val = emitExpr(apply.inputs()[0]);
+      // Special casing for list and tuple since isintance(x, list) and isinstance(x, tuple)
+      // does not accept List[int] / Tuple[int] like subscript type annotation in python
+      bool is_instance_val = false;
+      if (*type_name == "list" && val->type()->cast<ListType>()) {
```

Comment:
I don't think this handles something recursive, like
```
isinstance(data, (list, tuple))
```

(as in the constructor of PackedSequence)